lib/middleware/obscure-provenance.js

```diff
@@ -0,0 +1,12 @@
+
+module.exports = function create_device_obscurity (env) {
+  function obscure_device (req, res, next) {
+    if (res.entries && env.settings.obscureDeviceProvenance) {
+      for (var i = 0; i < res.entries.length; i++) {
+        res.entries[i].device = env.settings.obscureDeviceProvenance;
```

Comment:
Does this change the in-memory cache data entries? Smells like this should make a copy of the data before modifying it

Reply:
Good idea.